### ISSUE TITLE
Onnx upsample dynamic input shape export problem

### ISSUE THREAD

**Issue by Kepnu4** [alibaba/MNN#833]
# 平台(如果交叉编译请再附上交叉编译目标平台):  
# Platform(Include target platform as well if cross-compiling):  
python3.6

# Github版本:
# Github Version:
pip MNN-1.0.1

Export of Upsample OP with shape as input from ONNX is wrong.
Example (In pytorch):  
```
class Model(nn.Module):
    def __init__(self):
        super(TorchNetwork, self).__init__()
        self.layer = nn.Conv2d(3, 3, kernel_size=(1, 1), stride=(2, 2), padding=(0, 0), bias=False)
    
    def forward(self, inputs):
        l = self.layer(inputs)
        u = F.interpolate(l, inputs.shape[2:])
        return u
```

Gives following graph in ONNX:
![Screenshot from 2020-05-27 10-16-03](https://user-images.githubusercontent.com/2921717/82989284-1b349b80-a003-11ea-962d-1c31ac7752b7.png)

After export to MNN it looks like:
![model](https://user-images.githubusercontent.com/2921717/82989321-2be51180-a003-11ea-9238-ccd1b290f066.png)

So it gives wrong result on scales different from 2.0, for example for input shape [255x255].

**Comment by Kepnu4** [comment]
Well, i found that i need to pass `dynamic_axes={'input': [2, 3], 'output': [2, 3]}`  on export to skip input shape info. But then i got another error
```
Onnx slice must use the same dimensition
Convert Onnx's Op 17 , type = Slice, failed, may be some node is not const
```
I think it might be related to https://github.com/alibaba/MNN/issues/829 
Please close this issue if it's the case

**Comment by Kepnu4** [comment]
So i fixed that but with "hacks" which i don't know how to fix right.
First 
https://github.com/alibaba/MNN/blob/master/tools/converter/source/optimizer/merge/TensorConverterMerge.cpp#L231
In this optimizer, if input has more than one output (it's also shape op in my case) then input got duplicated. I commented this line to make it works.
Second
https://github.com/alibaba/MNN/blob/master/tools/converter/source/optimizer/onnxextra/OnnxUpsample.cpp#L56
I don't quite understand purpose of this check but it's always "False" in my case, so shape is never passed as variable. I also commented this line to make it works.
Third
https://github.com/alibaba/MNN/blob/master/tools/converter/source/optimizer/onnxextra/OnnxUpsample.cpp#L58
In ONNX Upsample second input is scales but in MNN it's sizes. It has to be converted somehow. For me i replaced this in onnx graph to make it works.

After this dirty hacks it worked fine. But obviously it's not a clean solution and i don't really understand how to fix it right. So it would be nice to hear from team about this issues.
